### PR TITLE
GDB-4768-add-siblings

### PR DESCRIPTION
**Pull request by desislava-hristova-ontotext**
Add siblings from each cell after all shared predicates/objects siblings and their nested triples.
Focus the new added empty cell.
Add proper turtle like triple ending with dot, comma and column.
Add tests of the new functionality.

### CODE REVIEW

File: src/app/main/mapper/iteration/iteration.component.ts [hunk 1]
```diff
@@ -564,4 +561,66 @@ export class IterationComponent extends OnDestroyMixin implements OnInit, AfterV
...
+      this.triples.splice(newTripleIndex, 0, new Triple(mapping.getSubject(), mapping.getPredicate(), undefined, mapping.isTypeProperty, false, false, OBJECT_SELECTOR));
+    }
+  }
+
+  private getRowEnding(triple: Triple) {
+    if (!triple.getSubject()) {
+      return '';
+    }
+    // Has type mappings
+    if (triple.isTypeProperty && triple.getSubject().getTypeMappings().length > 0) {
```
Comment (svilenvelikov) [inline src/app/main/mapper/iteration/iteration.component.ts:589]: Maybe extract some local variables like: subject, typeMappings, propertyMappings, etc. in order to make the conditions below shorter and readable.

File: cypress/test/add-siblings.spec.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+describe('Add siblings', () => {
+
+  beforeEach(() => {
+    PrepareSteps.prepareMoviesNamespacesAndColumns();
+    cy.route('GET', '/orefine/command/core/get-models/?project=123', 'fixture:add-siblings/mapping-model.json');
+    // Given I have created a mapping with multiple nested triples
+    // When I load the mapping
+    cy.visit('?dataProviderID=ontorefine:123');
+    // Then I expect to see 9 triples (+1 empty template)
```
[the remark this answers is not in the document]
Comment (desislava-hristova-ontotext) [reply cypress/test/add-siblings.spec.ts]: fixed

File: src/app/main/mapper/iteration/iteration.component.ts (hunk 1 above)
Comment (desislava-hristova-ontotext) [reply src/app/main/mapper/iteration/iteration.component.ts]: Done

**Comment by svilenvelikov** [comment]
You have a failing test for the siblings

**Comment by desislava-hristova-ontotext** [comment]
> You have a failing test for the siblings

As soon as I introduce tests, I start breaking them :-D Fixed that

**Pull request merged**